contents/tutorials/flutter-ab-tests.md

```diff
@@ -0,0 +1,358 @@
+
+Our app will have two screens:
+
+1. The first screen will have a button which takes you to a second screen.
+2. The second screen will either have a `red` or `green` background color, depending on whether the user is in the `control` or `test` variant of our A/B test. This screen will also have a button which captures an event when it's pressed. We'll use this event as our goal metric for our test.
+
+To set this up, first ensure the [Flutter extension for VS Code](https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter) is installed. Then, create a new app by opening the Command Palette in VS Code (`Ctrl/Cmd + Shift + P`), typing `flutter` and selecting `Flutter: New Project`. 
+
+Select `Empty Application` and name your app `flutter_ab_tests`. Then, replace your code in `lib/main.dart` with the following:
+
+```dart file=lib/main.dart
+import 'package:flutter/material.dart';
+import 'feature_screen_view.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'Flutter A/B Test App',
+      home: MainScreen(),
+    );
+  }
+}
+
+class MainScreen extends StatelessWidget {
+  const MainScreen({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Main Screen')),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            const Text('Hello, world!'),
+            ElevatedButton(
+              child: const Text('Go to Next Screen'),
+              onPressed: () {
+                Navigator.push(
+                  context,
+                  MaterialPageRoute(builder: (context) => const FeatureScreenView(isTestVariant: false)), // We update this later
+                );
+              },
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+```
+
+Lastly, in the `lib` directory, create a new file for our second screen called `feature_screen_view.dart`. Add the following code to it:
+
+```dart
+import 'package:flutter/material.dart';
+
+class FeatureScreenView extends StatelessWidget {
+  final bool isTestVariant;
+
+  const FeatureScreenView({Key? key, required this.isTestVariant}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: isTestVariant ? Colors.green : Colors.red,
+      body: Center(
+        child: ElevatedButton(
+          child: const Text('Click Me!'),
+          onPressed: () {
+            // Event capturing will go here
+          },
+        ),
+      ),
+    );
+  }
+}
+```
+
+Press **F5** and run the app in any emulator (we chose Android) to see your app in action.
+
+![Basic setup of the Flutter app](../images/tutorials/flutter-ab-tests/basic-app.mp4)
+
+## 2. Add PostHog to your app
```

Comment:
My comments on the other files apply to this file too since they are the same comments